vspec/__init__.py

```diff
@@ -236,6 +236,11 @@ def cleanup_flat_entries(flat_model):
 #
 # Delete parser-specific elements
 #
+# Parser metadata is cleaned in two steps. An initial step just after vspec files are parsed
+# and a second step after all model transformations have been performed
+#
+# First cleaning step - note that $file_name$ is not deleted as it is used for error messages
+#
```

Comment:
As of today `cleanup_deep_model` does not remove `$file_name$` (it is a core attribute and used in errors) and does not traverse "extended attributes". We could traverse extended attributes here, but then we would need a different handling to remove  `$file_name$` in them as well, if we want to remove all "metadata" in extended attributes